lib/firenest/replicated_state/remote.ex

```diff
@@ -0,0 +1,219 @@
+defmodule Firenest.ReplicatedState.Remote do
+  @moduledoc false
+
+  defstruct pending: %{}, clocks: %{}, clock: 0, tag: nil, deltas: nil, broadcast: nil
+
+  # TODO: some protocol for requesting more of a state, even from other nodes
+  # on first up, so we don't need to go to each node separately.
+
+  import Record
+  defrecord :deltas, max: nil, lowest: 0, store: %{}
+
+  def new(:ignore, broadcast, max_deltas)
+      when is_function(broadcast, 0) and is_integer(max_deltas) do
+    %__MODULE__{tag: :ignore, broadcast: broadcast, deltas: deltas(max: max_deltas)}
+  end
+
+  def clock(%__MODULE__{clock: clock}), do: clock
+
+  # Reconnections are dead until we have permdown
+  def up(%__MODULE__{clocks: clocks} = state, ref, clock) do
+    case clocks do
+      # Reconnection, try to catch up
+      %{^ref => old_clock} when clock > old_clock ->
+        {:catch_up, old_clock, state}
+
+      # Reconnection, no remote changes
+      %{^ref => old_clock} ->
```

Comment:
Any reason to prefer the sanity equality check over `%{^ref => ^clock}` ?

Reply:
Yes. It could fall into the following clauses.